rc/filetype/json.kak

```diff
@@ -32,11 +32,11 @@ provide-module json %(
 # Highlighters
 # ‾‾‾‾‾‾‾‾‾‾‾‾
 
-add-highlighter shared/json regions
-add-highlighter shared/json/code default-region group
-add-highlighter shared/json/string region '"' (?<!\\)(\\\\)*" fill string
-
-add-highlighter shared/json/code/ regex \b(true|false|null|\d+(?:\.\d+)?(?:[eE][+-]?\d*)?)\b 0:value
+add-highlighter shared/json group
+add-highlighter shared/json/value regions
+add-highlighter shared/json/value/string region '"' (?<!\\)(\\\\)*" fill string
+add-highlighter shared/json/value/ default-region regex \b(true|false|null|\d+(?:\.\d+)?(?:[eE][+-]?\d*)?)\b 0:value
+add-highlighter shared/json/key regex (?<!\\)("(?:[^"\\]|\\\\|\\")*")\s*: 1:keyword
```

Comment:
Not fully happy about that overwritting strings approach, I wonder if we could have something that matches language semantics more closely (probably using a recursive `json-object` shared highlighter), but at the same time it seems a proper solution would get really complex.

Reply:
You're right, this isn't a clean solution.

I was thinking about recursive highlighter previously as well. It could be possible splitting buffer in list and object regions, but I wasn't sure whether kakoune can handle recursive highlighters with regions within regions.

I can give the recursion a try, perhaps it won't be as complex as it seems.

Reply:
Any success with recursion ? Kakoune should handle recursive highlighting fine, if it does not thats a bug.

Reply:
When I tried to do recursive regions, kakoune refused to recurse once it detected recursive region highlighter definition. Here is the sanity check that trips: https://github.com/mawww/kakoune/blob/7438f23b9beddc42b6561fe8be3f953aff2f73b1/src/highlighters.cc#L1636

Reply:
That means your recursive highlighter is running in the same range as a parent one, so it is going to give the same result (resulting in an infinite loop), recursion is only going to work if you apply it on a sub-range of the buffer (in a region highlighter for example).